### ISSUE TITLE
How is docA.json generated ?

### ISSUE THREAD

**Issue by mcunha** [Lombra/elite-api-docs#8]
Trying to run the code myself so I can test changes locally if I wanted to add some extra linking support. However docA.json is missing from the repo and it's unclear how it's generated.

Is this something you're willing to share @Lombra ?

**Comment by Lombra** [comment]
Certainly! Though the entire process is very hacky, why I don't really want to commit it.

It stems from the fact that the documentation is distributed in rather inconvenient formats. What I do is I download the .doc version, open it in LibreOffice and export it as a HTML page. I then take that HTML and paste it into this page, which will then generate the `docA.json`.
[parse-html.zip](https://github.com/Lombra/elite-api-docs/files/12217685/parse-html.zip)

**Comment by mcunha** [comment]
Thank you for sharing, that's super helpful! I have a couple of things I want to try. I'll update you if I fail or can't continue, or send a PR if I succeed :)

**Comment by mcunha** [comment]
Heya Lombra, I've taken a solid stab at this. There's some major work done so I thought I'd update you before sending PR's.

This is a major expansion of the source code base so let me know which of these you'd like to see individual PR's for or if you'd prefer one chunky PR.

Added:
- Automated conversion from DOC > HTML > JSON > Markdown all done in build.js
- Added autolinking for all event names anywhere in the manual
- Added autolinking for all section number references anywhere in the manual
- Ensured all JSON blocks are pretty printed and syntax highlighted
- Added support for FDev style JSON comments in JSON
- Updated to latest MkDoc Material theme features:
  - Added support for light and dark themes [default is down to browser preferences]
  - Added seearch box for easy searching within manual
  - Added footer
  - Added permalinks to each header with easy one click clipboard copying
- Added typo/formatting corrections to:
  - Fix actual typos (or just missing words)
  - Fix broken references to event names or section references
  - Fix broken JSON preventing pretty printing
- Unified the language in the ChangeLog area

Started:
- Github Action workflows so we can just upload the new manual doc to the repo and it automatically generates fresh documentation, sends update to repo, which then triggers readthedocs

Pending:
- Support for inlining JSON inside a paragraph [not a JSON block]
- Support for inlining file paths

**Comment by Lombra** [comment]
Oh wow, that's impressive!

Auto linking and auto corrections (if I understand correctly) is something I wanted to do, but was probably never going to happen. I didn't want to commit to fixing too many of their mistakes, which I'm currently doing manually, so if that can be automated, that's great.

I think I would prefer individual PRs, but you could merge some of it if you like, such as the two autolinking features and content correction work.
